src/overlays/actors/ovl_En_Syateki_Wf/z_en_syateki_wf.c

```diff
@@ -64,60 +114,378 @@ const ActorInit En_Syateki_Wf_InitVars = {
+    GlobalContext* globalCtx = globalCtx2;
+    EnSyatekiWf* this = THIS;
+
+    if (this->actionFunc != func_80A20284) {
+        SkelAnime_Update(&this->skelAnime);
+    }
+
+    Actor_MoveWithGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 32.0f, 30.0f, 60.0f, 5);
+    this->actionFunc(this, globalCtx);
+
+    if ((this->actor.bgCheckFlags & 3) != 0) {
+        func_800BE3D0(&this->actor, this->actor.shape.rot.y, &this->actor.shape.rot);
+    } else {
+        Math_SmoothStepToS(&this->actor.shape.rot.x, 0, 1, 0x3E8, 0);
+        Math_SmoothStepToS(&this->actor.shape.rot.z, 0, 1, 0x3E8, 0);
+    }
+
+    if ((this->unk_2B4.base.acFlags & 2) || (this->unk_300.base.acFlags & 2) || (this->unk_34C.base.acFlags & 2)) {
+        this->unk_2B4.base.acFlags &= ~2;
+        this->unk_300.base.acFlags &= ~2;
+        this->unk_34C.base.acFlags &= ~2;
+        this->actor.colChkInfo.health -= 2;
+        if (this->actor.colChkInfo.health == 0) {
+            func_801A3098(NA_BGM_GET_ITEM | 0x900);
+            func_80A20858(this, globalCtx);
+        } else {
+            play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+            EffectSsExtra_Spawn(globalCtx, &this->actor.world.pos, &D_80A20EDC, &D_80A20EE8, 3, 0);
+        }
+    }
+
+    Collider_UpdateCylinder(&this->actor, &this->unk_2B4);
+    if ((this->actionFunc != func_80A20284) && (this->actionFunc != func_80A208F8) && (this->actor.draw != NULL)) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->unk_300.base);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->unk_2B4.base);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->unk_34C.base);
+        this->actor.focus.pos = this->actor.world.pos;
+        this->actor.focus.pos.y += 25.0f;
+    }
+
+    if (this->unk_2B0 == 0) {
+        if ((Rand_ZeroOne() < 0.2f) && ((globalCtx->gameplayFrames & 3) == 0) && (this->actor.colorFilterTimer == 0)) {
+            this->unk_2B0++;
+        }
+    } else {
+        this->unk_2B0 = (this->unk_2B0 + 1) & 3;
+    }
+}
+
+s32 EnSyatekiWf_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                 Actor* thisx) {
+    return false;
+}
+
+void EnSyatekiWf_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnSyatekiWf* this = THIS;
+    Vec3f sp18;
+
+    Collider_UpdateSpheres(limbIndex, &this->unk_34C);
+    if (limbIndex == 6) {
+        Matrix_MultiplyVector3fByState(&D_80A20FD0, &sp18);
+        this->unk_300.dim.pos.x = sp18.x;
+        this->unk_300.dim.pos.y = sp18.y;
+        this->unk_300.dim.pos.z = sp18.z;
+    }
+}
+
+void EnSyatekiWf_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnSyatekiWf* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(D_80A20FDC[this->unk_2B0]));
```

Comment:
Surely these are eye textures, therefore.

Reply:
I didn't want name this unless I looked at the object, but since you brought it up, I checked the object out and yes, these are eye textures.

Reply:
At this point it's pretty obvious, since they're the same as used in the main Wf file. And, since I know that Wolfos blink, it was basically the only option. Hmm, I wonder how similar the object is to OoT.